staging/src/k8s.io/component-base/config/testing/roundtrip.go

```diff
@@ -90,3 +92,89 @@ func GetRoundtripTestCases(scheme *runtime.Scheme, disallowMarshalGroupVersions
 	}
 	return cases
 }
+
+func runRoundTripTests(t *testing.T, tests []TestCase, codecs serializer.CodecFactory) {
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			object := decodeYAML(t, tc.in, tc.codec)
+
+			roundTrip(t, tc.codec, object)
+
+			data, err := runtime.Encode(tc.codec, object)
```

Comment:
Should we move matchOutputFile into roundTrip, so we don't have to re-encode here?